Controller/TokenController.php

```diff
@@ -37,9 +55,25 @@ public function __construct(OAuth2 $server)
      */
     public function tokenAction(Request $request)
     {
-        try {
-            return $this->server->grantAccessToken($request);
-        } catch (OAuth2ServerException $e) {
+        try
+        {
+            $response = $this->server->grantAccessToken($request);
+
+            $data = json_decode($response->getContent(), true);
+
+            $storage = $this->container->get('fos_oauth_server.storage');
+            /* @var $storage \FOS\OAuthServerBundle\Storage\OAuthStorage */
+            $accessToken = $storage->getAccessToken($data[OAuth2::TOKEN_PARAM_NAME]);
+            /* @var $accessToken \FOS\OAuthServerBundle\Entity\AccessToken */
+
+            $this->container->get('event_dispatcher')->dispatch(
+                OAuthTokenEvent::POST_ACCESS_TOKEN_GRANT,
+                new OAuthTokenEvent($accessToken)
+            );
+
+            return $response;
+        }
+        catch (OAuth2ServerException $e) {
```

Comment:
No need to apply this ``try ... catch`` to the entire function.
just apply it on the first line.